### DISCUSSION TITLE
Why do we need Turbomodules when writing with C++?

### DISCUSSION THREAD

**Discussion by shamilovtim** [callstack/react-native-builder-bob#356] [Q&A] [answered]
Yesterday I went about creating a new library with bob.

I wanted to go C++ and JSI. I selected the new architecture + backwards compat option, then selected C++. What I noticed was that this created a Turbomodule for my library. However, I thought that one of the strengths of using a JSI+JNI module was that it could be **alternative** to needing to write for either TurboModules vs. Native Modules. So we'd get compat with both old and new architecture out of the box. 

As such as I'm confused and looking for an answer to this. Should there be three options in the initial CLI prompt?
1. native module
2. turbo module
3. JSI / JNI C++ module (compatible with both)

If I just want a JSI module is there something I can currently select in the CLI to achieve my desired outcome? 
What are the pros/cons of the JSI module also being a turbomodule (which is what it currently seems I'm stuck with)? 

The core of my library is currently written in plain C so I was hoping to skip any marshaling to Java or Swift entirely and go C-> React Native. I'm concerned that adding the TurboModule layer is just extra baggage.

Thank you in advance!

**Comment by satya164** [comment, marked as answer]
There is no pure JNI template because I don't know C/C++ so don't have enough knowledge to write a decent template. Pull requests are welcome.

**Reply by shamilovtim** [reply]
If we end up needing a pure C++ impl I would be willing to create a JNI template. We're actually leaning toward Java + Obj-C now so might not need it after all.
